File: app/src/main/java/com/zskjprojectj/andouclient/adapter/PoiItemListAdapter.java
package com.zskjprojectj.andouclient.adapter;

import com.amap.api.services.core.PoiItem;
import com.chad.library.adapter.base.BaseViewHolder;
import com.zhuosongkj.android.library.adapter.BaseAdapter;
import com.zskjprojectj.andouclient.R;


public class PoiItemListAdapter extends BaseAdapter<PoiItem> {

    private int selectPosition = -1;

    public PoiItemListAdapter() {
        super(R.layout.mapresult_view);
    }

    public void setSelectPosition(int position) {
        this.selectPosition = position;
        notifyDataSetChanged();
    }

    public int getSelectPositon() {
        return selectPosition;
    }

    @Override
    protected void convert(BaseViewHolder helper, PoiItem item) {
        helper.setChecked(R.id.checkBox, helper.getAdapterPosition() == selectPosition)
                .setText(R.id.tv_title, item.getTitle())
                .setText(R.id.tv_message, item.getProvinceName()
                        + item.getCityName()
                        + item.getAdName()
                        + item.getSnippet());
    }
}

File: app/src/main/java/com/zskjprojectj/andouclient/adapter/PoiItemListAdapter.kt
package com.zskjprojectj.andouclient.adapter

import android.graphics.Color
import com.amap.api.services.core.PoiItem
import com.chad.library.adapter.base.BaseViewHolder
import com.zhuosongkj.android.library.adapter.BaseAdapter
import com.zskjprojectj.andouclient.R

class PoiItemListAdapter : BaseAdapter<PoiItem>(R.layout.mapresult_view) {
    override fun convert(helper: BaseViewHolder, item: PoiItem) {
        helper.itemView.isSelected = helper.adapterPosition == 0
        helper.setText(R.id.tv_title, item.title)
                .setTextColor(R.id.tv_title,
                        if (helper.adapterPosition == 0)
                            Color.parseColor("#61D2AE")
                        else
                            Color.parseColor("#ff333333"))
                .setText(R.id.tv_message, item.provinceName
                        + item.cityName
                        + item.adName
                        + item.snippet)
    }
}